autopxd/__init__.py

```diff
@@ -34,11 +36,92 @@ def ensure_binary(s, encoding="utf-8", errors="strict"):
     raise TypeError(f"not expecting type '{type(s)}'")
 
 
+def _find_cl():
+    """Use vswhere.exe to locate the Microsoft C compiler."""
+    host_platform = {
+        "X86": "X86",
+        "AMD64": "X64",
+        "ARM64": "ARM64",
+    }.get(platform.machine(), "x86")
+    build_platform = {
+        "X86": "x86",
+        "AMD64": "x64",
+        "ARM64": "arm64",
+    }.get(platform.machine(), "x86")
+    cmd = [
+        os.path.expandvars(r"%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe"),
```

Comment:
Should it be `ProgramFiles({host_platform})`?

```suggestion
        os.path.expandvars(fr"%ProgramFiles({host_platform})%\Microsoft Visual Studio\Installer\vswhere.exe"),
```

Reply:
Actually, the case where this needs to change is when running on a 32-bit OS (which is incredibly rare these days, but I'll add it anyway).